### ISSUE TITLE
Unable to reproduce results on cora

### ISSUE THREAD

**Issue by Frostland12138** [Suchun-sv/SGDD#9]

**Comment by Frostland12138** [comment]
I ran your code by 
` python train_SGDD.py  --dataset cora  --nlayers=2 --beta 0.1 --r=1 --gpu_id=0`
I got training loss to be 'nan' for many times;
`=== picking the best model according to the performance on validation ===
Train set results: loss= 1.4607 accuracy= 0.9286
Test set results: loss= 1.7068 accuracy= 0.7390
Train/Test Mean Accuracy: [array([0.93333333, 0.72666667]), array([0.00673435, 0.06306786])]`
for one time.

I would greatly appreciate your assistance in identifying the issue to replicate your work.
